dissect/target/plugins/os/windows/regf/usb.py

```diff
@@ -1,29 +1,38 @@
+from __future__ import annotations
+
+import re
 import struct
+from typing import Iterator
 
 from dissect.util.ts import wintimestamp
 
-from dissect.target.exceptions import RegistryValueNotFoundError, UnsupportedPluginError
+from dissect.target.exceptions import (
+    RegistryKeyNotFoundError,
+    RegistryValueNotFoundError,
+    UnsupportedPluginError,
+)
 from dissect.target.helpers.record import TargetRecordDescriptor
-from dissect.target.plugin import Plugin, export, internal
+from dissect.target.helpers.regutil import VirtualKey
+from dissect.target.plugin import Plugin, export
 
-UsbRegistryRecord = TargetRecordDescriptor(
+USBRegistryRecord = TargetRecordDescriptor(
     "windows/registry/usb",
     [
-        ("string", "device_type"),
+        ("string", "type"),
         ("string", "serial"),
-        ("string", "vid"),
-        ("string", "pid"),
-        ("string", "rev"),
-        ("string", "containerid"),
+        ("string", "container_id"),
         ("string", "vendor"),
         ("string", "product"),
-        ("string", "version"),
-        ("string", "friendlyname"),
+        ("string", "revision"),
+        ("string", "friendly_name"),
         ("datetime", "first_insert"),
         ("datetime", "first_install"),
         ("datetime", "last_insert"),
         ("datetime", "last_removal"),
-        ("string", "info_origin"),
+        ("string[]", "volumes"),
+        ("string[]", "mounts"),
+        ("string[]", "users"),
```

Comment:
I'm fine with this being in this record too, but ideally you may also have a separate export that yields individual records for each user, where you have the `_user` set to the user record and it linked by serial. Same for volumes and mounts.

However that's probably a bit bigger as it probably requires this to be converted to a namespace plugin, can create a separate ticket for that.

Reply:
#779